File: ai-interview-frontend/src/api/positionAgent.js
import api from './request'

/**
 * 运行岗位匹配 Agent
 * @param {{ resume_id: number, target_direction?: string }} payload
 */
export function matchPositions(payload) {
  // Agent 调用比较慢，超时设为 3 分钟
  return api.post('/position-agent/match', payload, { timeout: 180000 })
}

/**
 * 从 Agent 推荐结果直接启动专项模拟面试
 * @param {{ resume_id: number, position_tag: string, difficulty?: string, total_questions?: number }} payload
 */
export function startInterviewFromAgent(payload) {
  return api.post('/position-agent/start-interview', payload, { timeout: 120000 })
}

File: app/schemas/client/position_agent.py
from pydantic import BaseModel, Field
from typing import Optional


class PositionMatchRequest(BaseModel):
    resume_id: int = Field(..., description="已解析完成的简历 ID")
    target_direction: Optional[str] = Field(None, description="用户期望方向，如 'Python 后端'（可选，会传给 Agent 作参考）")


class StartInterviewFromAgentRequest(BaseModel):
    resume_id: int
    position_tag: str
    difficulty: Optional[str] = None
    total_questions: Optional[int] = None
